protractor/e2e/integration/blog-page.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { BlogPo } from '../support/blog.po';
+import { SecondPo } from '../support/second.po';
+import {PaginationPo} from "../support/pagination.po";
+
+describe("blog page", () => {
+    const blogPage = new BlogPo();
+    const secondPage = new SecondPo();
+    const paginationElement = new PaginationPo();
+    const numberOfArticles: number = 20;
+
+    it('navigate from second page to blog', async () => {
+        await secondPage.navigateTo();
+        await secondPage.waitForElementVisible(secondPage.blogHeaderLink);
+        await secondPage.blogHeaderLink.click();
+        await blogPage.waitForElementVisible(blogPage.posts.first());
+        expect(await blogPage.getCurrentUrl()).toContain(blogPage.pageUrl);
+    });
+
+    it('user see 20 articles with header, description and link', async () => {
+        await secondPage.navigateTo();
+        await secondPage.waitForElementVisible(secondPage.blogHeaderLink);
+        await secondPage.blogHeaderLink.click();
+        await blogPage.waitForElementVisible(blogPage.posts.first());
+        expect(await blogPage.posts.count()).toBe(numberOfArticles);
+        const postsCount = await blogPage.posts.count();
+        for (let i = 0; i < postsCount; i++) {
+            let header = blogPage.postHeaders.get(i);
+            let entry = blogPage.postEntries.get(i);
+            await blogPage.scrollToElement(entry);
+            expect(await header.getText().toString().length).toBeGreaterThan(1, "headers is not empty");
+            expect(await entry.getText().toString().length).toBeGreaterThan(1, "description is not empty");
+        }
```

Comment:
you can use each function from protractor for this goal:
blogPage.posts.each((post, index) => {
expect(post.isDisplayed()).toBeTruthy();
expect(blogPage.postHeaders.get(index).isDisplayed()).toBeTruthy();
expect(blogPage.postHeaders.get(index).isDisplayed()).toBeTruthy();
})